strawberryfields/engine.py

```diff
@@ -609,15 +610,63 @@ def run_async(self, program: Program, *, compile_options=None, **kwargs) -> Job:
         """
         # get the specific chip to submit the program to
         compile_options = compile_options or {}
+        recompile = compile_options.pop("recompile", False)
```

Comment:
should `recompile` be part of compile options? It feels like it should be a separate keyword argument (since it is not passed to `Program.compile`, as the docstring says)